cola/algorithms/cg.py

```diff
@@ -78,6 +68,9 @@ def run_batched_cg(A, b, x0, max_iters, tol, preconditioner, pbar):
     mult = xnp.norm(b, axis=-2, keepdims=True)
     b_norm = do_safe_div(b, mult, xnp=xnp)
     init_val = initialize(A=A, b=b_norm, preconditioner=preconditioner, x0=x0, xnp=xnp)
+    _, _, r0, *_ = init_val
+    init_tol = tol
+    tol = tol * xnp.norm(r0, axis=-2, keepdims=True)
```

Comment:
Looks like a good way to do it to me, only thing I might do is compute `xnp.norm(r0+b0, axis=-2, keepdims=True)` in case `x0` is already close to the solution in which case the residual will be much smaller than the largest eigenvalue of `A`

Reply:
Yeah good point

Reply:
Since ||a + b || <= ||a|| + ||b||. And ||b|| = 1 I'm gonna use `tol = tol * xnp.norm(r0, axis=-2, keepdims=True) + tol`.

Reply:
That way I don't have to worry of the small case, but possible were r0 and b affect each other